### PR TITLE
[ENH] Pyjanitor for PySpark

### PR DESCRIPTION
**Pull request by zjpoh**
# PR Description

* Added pyspark backend at `janitor.spark.backend`. Using the same naming convention as `janitor`, i.e. `spark` instead of `pyspark`.
* Like `janitor` import `janitor.spark.functions` when importing `janitor.spark`.
* Added a `clean_names` function at `janitor.spark.functions`. Also demonstrate that we can reuse some functions in `janitor`.

**This PR addresses #504.**

# PR Checklist

Please ensure that you have done the following:

1. [x] PR in from a fork off your branch. Do not PR from `<your_username>`:master, but rather from `<your_username>`:<branch_name>.

2. [x] If you're not on the contributors list, add yourself to `AUTHORS.rst`.

3. [ ] Add a line to `CHANGELOG.rst` under the latest version header (i.e. the one that is "on deck") describing the contribution.
    - Do use some discretion here; if there are multiple PRs that are related, keep them in a single line.

## Quick Check

To do a very quick check that everything is correct, follow these steps below:

- [x] Run the command `make check` from pyjanitor's top-level directory. This will automatically run:
    - black formatting
    - pycodestyle checking
    - running the test suite
    - docs build
    
Once done, please check off the check-box above.
    
If `make check` does not work for you, you can execute the commands listed in the Makefile individually.

## Code Changes

If you are adding code changes, please ensure the following:

- [x] Ensure that you have added tests.
- [x] Run all tests (`$ pytest .`) locally on your machine.
    - [ ] Check to ensure that test coverage covers the lines of code that you have added.
    - [x] Ensure that all tests pass.

## Documentation Changes

If you are adding documentation changes, please ensure the following:

- [x] Build the docs locally.
- [ ] View the docs to check that it renders correctly.

# Relevant Reviewers

Please tag maintainers to review.

- @ericmjl

### CODE REVIEW

File: tests/spark/functions/test_clean_names.py [hunk 1]
```diff
@@ -0,0 +1,26 @@
+import pytest
+import pandas as pd
+from pyspark.sql import SparkSession
+from sklearn.datasets import load_iris
+
+import janitor.spark
+
+
+@pytest.fixture(scope="module")
+def df():
```
Comment (zjpoh) [inline tests/spark/functions/test_clean_names.py]: Can't figure out how to use hypothesis with spark. Any help would be appreciated.

**Comment by zjpoh** [comment]
@ericmjl @anzelpwj I haven't completed all the checklist yet but I hope this can kickstart some more discussions.

File: tests/spark/functions/test_clean_names.py (hunk 1 above)
Comment (ericmjl) [reply tests/spark/functions/test_clean_names.py]: No need to worry about the use of Hypothesis. It's there mainly to generate lots of examples, and help with catching edge-cases that I might not have thought about.

**Comment by ericmjl** [comment]
Looks great, thanks a ton, @zjpoh! Looking forward to seeing this grow into something!
